sources/web/datalab/polymer/components/item-list/item-list.ts

```diff
@@ -62,39 +65,103 @@ class ItemListElement extends Polymer.Element {
         value: function(): Array<Object> {
           return [];
         },
+        observer: '_rowsChanged',
       },
       columns: {
         type: Array,
         value: function(): Array<string> {
           return [];
         }
+      },
+      _selectedElements: {
+        type: Array,
+        value: function(): Array<Object> {
+          return [];
+        }
       }
     }
   }
 
   /**
-   * on row click, check the click target, if it's the checkbox, add it to
-   * the selected rows, otherwise select it only
+   * Returns list of currently selected elements. This list keeps the actual
+   * HTML elements, which can then be used to get their indices, whereas the
+   * opposite is not directly possible.
+   */
+  getSelectedElements() {
+    return this._selectedElements;
+  }
+
+  /**
+   * Returns list of indices for the currently selected elements.
+   */
+  getSelectedIndices() {
+    return this._selectedElements.map(element => {
+      return this.$.list.indexForElement(element);
+    });
+  }
+
+  /**
+   * When the list of rows is refreshed, no items should be selected.
+   */
+  _rowsChanged() {
+    this._selectedElements = [];
+  }
+
+  /**
+   * On row click, checks the click target, if it's the checkbox, adds it to
+   * the selected rows, otherwise selects it only.
+   * This method also maintains the _selectedElements list
    */
   _rowClicked(e: MouseEvent) {
     const target = <HTMLDivElement>e.target;
     const index = this.$.list.indexForElement(target);
+    const rowElement = this._getRowElementFromChild(target);
 
-    // if the clicked element is the checkbox, we're done, the checkbox already
-    // toggles selection (see the dom-repeat template)
-    // otherwise, select this element, unselect all others
+    // If the clicked element is the checkbox, we're done, the checkbox already
+    // toggles selection.
+    // Otherwise, select this element, unselect all others.
     if (target.tagName !== 'PAPER-CHECKBOX') {
       for (let i = 0; i < this.rows.length; ++i) {
         this.set('rows.' + i + '.selected', false);
       }
       this.set('rows.' + index + '.selected', true);
+
+      // This is now the only selected element.
+      if (rowElement)
```

Comment:
I had to look at this for a while before figuring it out. rowElement is null or undefined when the clicked element is the checkbox, right? If so, please add a comment somewhere explaining that. If I'm wrong, please add whatever other comment is necessary to explain.